recipes/gmp/all/conanfile.py

```diff
@@ -69,19 +69,25 @@ def package(self):
         with tools.chdir(self._source_subfolder):
             autotools = self._configure_autotools()
             autotools.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
         tools.rmdir(os.path.join(self.package_folder, "share"))
-        # remove la files
-        for la_name in ['libgmp.la', 'libgmpxx.la']:
-            la = os.path.join(self.package_folder, "lib", la_name)
-            if os.path.isfile(la):
-                os.unlink(la)
+        os.unlink(os.path.join(self.package_folder, "lib", "libgmp.la"))
+        if self.options.enable_cxx:
+            os.unlink(os.path.join(self.package_folder, "lib", "libgmpxx.la"))
 
     def package_id(self):
         del self.info.options.run_checks  # run_checks doesn't affect package's ID
 
     def package_info(self):
+        super_component = "libgmp"
+        self.cpp_info.components["libgmp"].libs = ["gmp"]
+        self.cpp_info.components["libgmp"].names["pkg_config"] = ["gmp"]
         if self.options.enable_cxx:
-            self.cpp_info.libs = ["gmpxx"]
-        self.cpp_info.libs.append("gmp")
-        self.cpp_info.names["cmake_find_package"] = "GMP"
-        self.cpp_info.names["cmake_find_package_multi"] = "GMP"
+            super_component = "gmpxx"
+            self.cpp_info.components["gmpxx"].libs = ["gmpxx"]
+            self.cpp_info.components["gmpxx"].requires = ["libgmp"]
+
+        # Make sure the GMP cmake_find_package generator knows about both libgmp and gmpxx
+        self.cpp_info.components[super_component].names["cmake_find_package"] = "GMP"
```

Comment:
So with this "super_component", this means that the ``gmp::GMP`` target will be different depending on the ``enable_cxx`` option? That looks a bit weird to me, but I have no idea how it is made so...

Reply:
This indeed seems a bit weird.
I'll change it to `GMP` for `gmp` and `GMPXX` for `gmpxx`.

I don't think people usually the `gmp` library when searching for `gmp`.